### ISSUE TITLE
Ability to customise primary key

### ISSUE THREAD

**Issue by craicoverflow** [aerogear/graphback#551]
## Motivation

The current primary key rules are very strict:

### graphql-migrations

- Can only use `ID` scalar, which maps to Knex `increments` (an auto-incrementing integer).
- Field name must be `id`.

### Graphback

- Can only use `ID` scalar, which maps to Knex `increments` (an auto-incrementing integer).
- Field name must be `id`.
- Generation engine does not allow ability to pass custom ID from server/client.

## Strategy

A new strategy will need to be considered to increase the flexibility of primary key customisation.

### graphql-migrations

Issues arise when changing a field with Knex to `increments` from something else.

This creates an integer **primary key**

```
table.integer('id').primary();
```

This creates an auto-incrementing **primary key**. Note that is does not require `.primary()`.

```
table.increments();
```

TL;DR this needs further investigation

This difference makes it a challenge since `.increments` always assumes a primary key, so Knex will try to create two primary keys.

### Graphback

This should be more straightforward. When a developer uses the `ID` scalar then Graphback will not pass the responsibility to provide the ID to the server and client. If the developer wants to set these they can use `Int`, `String` etc.

**Comment by wtrocki** [comment]
I see this as very low priority over the other changes. The problem we have here can be resolved by better validation.

**Comment by wtrocki** [comment]
Adding documentation should be enough for the moment
